### PR TITLE
Handle schemas as common attribute

### PR DESCRIPTION
**Pull request by MelinaGirl**
## Purpose
According to the specification the "schemas" attribute is a common attribute so there is no need to handle it separately in class AbstractSCIMObject. Also decoding becomes more simple and straight forward.

## Approach
We added the schemas as a MultiValuedAttribute to the attributeList of AbstractSCIMObject and adapted the setter and getter methods for schemas to work on the attributeList without changing the signatures of these methods.

### CODE REVIEW

**Comment by ashensw** [comment]
@MelinaGirl The build([Travis-ci](https://travis-ci.org/wso2/charon/builds/458347877?utm_source=github_status&utm_medium=notification)) is failing for this PR. Can you please check?

**Comment by MelinaGirl** [comment]
@ashensw  Done :)

**Comment by MelinaGirl** [comment]
Ping! Please review :)

**Comment by sebastianloesch** [comment]
Hey folks, any progress on this PR? Can we support you? Best regards, Sebastian

File: modules/charon-core/src/main/java/org/wso2/charon3/core/schema/SCIMConstants.java [hunk 1]
```diff
@@ -66,6 +66,7 @@
     public static class CommonSchemaConstants {
 
         public static final String SCHEMAS = "schemas";
+        public static final String SCHEMAS_URI = "urn:ietf:params:scim:schemas:core:2.0:schemas";
```
Comment (ashensw) [inline modules/charon-core/src/main/java/org/wso2/charon3/core/schema/SCIMConstants.java]: @MelinaGirl I guess the correct URI should be `urn:ietf:params:scim:schemas:core:2.0:Schema`?

Comment (MelinaGirl) [reply modules/charon-core/src/main/java/org/wso2/charon3/core/schema/SCIMConstants.java]: @ashensw No, in this case the URI is meant as a URI for the AttributeSchema "SCHEMAS", while your suggestion is refering to the URI for the schema resource (Specified in RFC7643 Section 7). Here it is for the attribute called "schemas", which is required for every resource to have (Specified in RFC7643 Section 3). Currently this required attribute is not handled the way all other attributes are handled, but by a separate "schemasList" attribute in the class AbstractSCIMObject, which causes inconsistencies.

File: modules/charon-core/src/main/java/org/wso2/charon3/core/objects/AbstractSCIMObject.java [hunk 2]
```diff
@@ -46,9 +45,6 @@
     /**Collection of attributes which constitute this resource.*/
     protected Map<String, Attribute> attributeList = new HashMap<String, Attribute>();
 
-    /**List of schemas where the attributes of this resource, are defined.*/
-    protected List<String> schemaList = new ArrayList<String>();
```
Comment (IsuraD) [inline modules/charon-core/src/main/java/org/wso2/charon3/core/objects/AbstractSCIMObject.java]: It is not recommended to remove protected/public variables. Those variables might be already used.

**Comment by DMHP** [comment]
@MelinaGirl , Thanks for your contribution. Closing the PR until the code review comments are addressed. Please reopen the PR after addressing the comments.

**Pull request closed without merging**